src/deep-security/lib/Security.js

```diff
@@ -148,16 +83,17 @@ export class Security extends Kernel.ContainerAware {
    * @returns {Token}
    */
   login(providerName, userToken, userId, callback) {
-    // check if providerName is defined
-    this.getLoginProviderConfig(providerName);
+    let identityProvider = new IdentityProvider(this._identityProviders, providerName, userToken, userId);
 
     let TokenImplementation = this._localBackend ? LocalToken : Token;
 
-    this._token = new TokenImplementation(this._identityPoolId, providerName, userToken, userId);
+    this._token = TokenImplementation.createFromIdentityProvider(this._identityPoolId, identityProvider);
 
     this._token.userProvider = this.userProvider;
 
-    this._token.getCredentials(callback);
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['credentials' is defined but never used.](https://www.codacy.com/app/MitocGroup/deep-framework/file/1371453481/issues/source?bid=2895475&fileBranchId=2973909#l94)